### ISSUE TITLE
why ssr module registration was removed from v16

### ISSUE THREAD

**Issue by tcstory** [vuejs/vue-loader#1887]
### Version
16.8.1

### Reproduction link
[github.com](https://github.com/vuejs/vue-loader/tree/next)

### Steps to reproduce
i have no idea why the functionality was removed. it's very important for ssr rendering.

vitejs/plugin-vue and vue-loader are both plugins, why vitejs/plugin-vue has this functionality but vue-loader doesnt

### What is expected?
ssr module registration should be support

### What is actually happening?
ssr module registration was removed

**Comment by sodatea** [comment]
Please be more specific. I don't understand what you mean by "ssr module registration"

**Comment by phoenix-ru** [comment]
Hi @sodatea. I've implemented critical CSS collection in https://github.com/phoenix-ru/vue-loader.

However, my solution assumes that `vue-style-loader` is used (as with v15). Can you verify if that's acceptable or v16 dropped `vue-style-loader` for a reason?

If that solution is fine, I can open a PR and polish my solution

**Comment by BlackCubeNo99** [comment]
Does anyone have a work around?

**Comment by BlackCubeNo99** [comment]
> Oh, that was a planned but not yet implemented feature. As you can see, the [SSR tests](https://github.com/vuejs/vue-loader/blob/820d23cbf16013dae894e0d84ed9da6e58a37584/test/ssr.spec.ts) are still listed as TODOs.
> 
> Thanks for bringing this up. I'll see if I can implement it when I have time.
> 
> Contributions are also welcome.

Does anyone have a work around?

**Comment by dmitryuk** [comment]
Hello, any plans?

**Comment by OskarLebuda** [comment]
> Hello, any plans?

@dmitryuk  I don't think so. There is a open PR since August by @niksy https://github.com/vuejs/vue-loader/pull/2083
But @yyx990803, @haoqunjiang  or any other contributors doesn't care about it. The fork is the only way I think. 

It will be only worst. The rolldown comming up, so I think they will drop support for Webpack any time soon. Last commit in this repo was 10 months ago. 

We've made a fork for our purposes, add patch by niks and seams to work fine for us.

**Comment by niksy** [comment]
This is really unfortunate becuase Vue loader will probably still be actively used because of Rspack and it’s Vue support.

The other solution (and probably best approach) can be [unplugin-vue](https://github.com/unplugin/unplugin-vue), but it’s [missing HMR support for Webpack/Rspack](https://github.com/unplugin/unplugin-vue/issues/162).
